### DISCUSSION TITLE
GC-FID data: automatinc peak detector / integrator not working

### DISCUSSION THREAD

**Discussion by AnnaClo** [OpenChrom/openchrom#470] [Q&A] [answered]
I have data detected simultneously on gc ms and fid. OpenChrom detects and integrates properly the ms data, but not the fid peaks. For some reason some of the peaks result integrated only halfaway - see screenshots. 
The other half of the peak is not quantified, not even as a separate peak. Before peak integration, I can see that the dotted line reaches the top of the peak but doesn't climb down the other half, so I guess the problem is peak detection. 
Any idea what could be the fix? 
I managed to get some peaks right by forcing 'optimize the baseline (VV) in peak detection, but the problem still appears for a handful of peaks per chromatogram. This isssue is not there at all in MS data. 

Here below a peak detected and integrated with the same procedure, from ms data, and from fid data respectively.

![image](https://github.com/OpenChrom/openchrom/assets/140986173/4c8009d9-549a-4f1a-b623-de101d56b254)

![image](https://github.com/OpenChrom/openchrom/assets/140986173/9bb3af95-8a22-4818-a4e6-68755dc22369)

**Comment by eselmeister** [comment]
With GC-FID, smoothing might be required first. Could you provide the displayed data set?

**Comment by AnnaClo** [comment]
Yes sure, here is a link to the same dataset. The peaks are very tiny compared to the solvent, between 13 and 27 mins. https://we.tl/t-AfNfUjLLBc

**Comment by eselmeister** [comment]
Thanks for the file. I would recommend to run the following steps on the GC-FID chromatogram:

1) Remove the solvent and first big peak via Chromatogram Filter > Scan Remover
2) Set a baseline via Baseline Detector > SNIP (Iterations: 250, Size: 9)
3) Remove the baseline via Chromatogram Filter > Chromatogram Baseline Subtract Filter
4) Smooth the data Chromatogram Filter > Savitzgy-Golay (Order: 2, Width: 9)
5) Detect peak via Peak Detector > First Derivative (Include Background: yes, Use Noise Segments: yes, Optimize Baseline: yes)


![Screenshot from 2023-07-31 07-42-33](https://github.com/OpenChrom/openchrom/assets/686671/4d6e79e1-6a41-4bfd-9659-7eeb184382fc)

**Comment by AnnaClo** [comment]
Thank you for indicating the steps, I used them and I managed to integrate peaks well. 
I only have a minor issue with the removal of the first big peak, the scan remover doesn't take it away. Did you use any specific setting?

**Comment by eselmeister** [comment, marked as answer]
@AnnaClo  Watch out: there are two big peaks, one at ~3.2 Minutes, the other at ~6.1 Minutes:

![Screenshot from 2023-08-01 09-53-00](https://github.com/OpenChrom/openchrom/assets/686671/d2998e77-0506-4727-9733-3c5246be3b08)
![Screenshot from 2023-08-01 09-53-15](https://github.com/OpenChrom/openchrom/assets/686671/6b0bcc3c-4fa2-47b9-8478-72f11cec87cb)
![Screenshot from 2023-08-01 09-53-36](https://github.com/OpenChrom/openchrom/assets/686671/0ed2970d-7433-49e3-ac2e-195daeaca549)
![Screenshot from 2023-08-01 09-53-42](https://github.com/OpenChrom/openchrom/assets/686671/c04dc53d-7576-4b14-8777-b8309be03a59)

**Comment by AnnaClo** [comment]
Oh ok I got it! Thank you!

**Reply by AbubakarsadiqT** [reply]
Please how do I view peaks on Openchrom? I have imported data but no peaks appeared.

**Reply by eselmeister** [reply]
Please provide a chromatogram file. I'll then show you how to do it.

**Reply by AbubakarsadiqT** [reply]
Okay. Here is the file. 
[Openchromjane.csv](https://github.com/user-attachments/files/19719916/Openchromjane.csv)

**Reply by eselmeister** [reply]
It seems to be not a valid chromatogram. Normally, chromatograms have ~5000 scans and more.

![image](https://github.com/user-attachments/assets/812c5fca-1779-4e21-9ea5-bac435b2b1c3)
